jobsdb/migration.go

```diff
@@ -657,13 +657,17 @@ func (jd *Handle) checkIfMigrateDS(ds dataSetT) (
 		return true, false, recordsLeft, nil
 	}
 
-	isSmall := float64(totalCount) < jd.conf.migration.jobMinRowsMigrateThres()*float64(jd.conf.MaxDSSize.Load())
+	isSmallDS := float64(totalCount) < jd.conf.migration.jobMinRowsMigrateThres.Load()*float64(jd.conf.MaxDSSize.Load())
 
-	if float64(delCount)/float64(totalCount) > jd.conf.migration.jobDoneMigrateThres() {
-		return true, isSmall, recordsLeft, nil
+	if float64(delCount)/float64(totalCount) > jd.conf.migration.jobDoneMigrateThres.Load() {
+		return true, isSmallDS, recordsLeft, nil
 	}
 
-	if isSmall {
+	if isSmallDS {
+		return true, true, recordsLeft, nil
+	}
+
+	if float64(recordsLeft) < jd.conf.migration.jobMinRowsLeftMigrateThres.Load()*float64(jd.conf.MaxDSSize.Load()) {
```

Comment:
With this condition, I think we can remove the above `isSmall` check and its config(this new condition becomes `isSmall`).
Too many conditions present might increase cognitive load.

Reply:
Yes, we may remove isSmall and its config in the next release after verifying behaviour of the new config in production 👍